engine/servermaster/serverutil/watch_executors.go

```diff
@@ -64,6 +63,7 @@ func WatchExecutors(ctx context.Context, watcher executorWatcher, user executorI
 	if err != nil {
 		return errors.Annotate(err, "watch executors")
 	}
+	defer updates.Close()
```

Comment:
Why we don't trigger this bug before?

Reply:
If you meant "Why had this bug never been triggered before", then it's probably because the error handling path had never been taken.